### PR TITLE
aiven patches on top of upstream version 1.5

### PR DESCRIPTION
**Pull request by mhoffm-aiven**

### CODE REVIEW

File: src/query/api/v1/handler/influxdb/write.go [hunk 1]
```diff
@@ -165,8 +165,13 @@ func (ii *ingestIterator) Next() bool {
 				for _, tag := range ptags {
 					name := make([]byte, len(tag.Key))
 					copy(name, tag.Key)
+					// tag.Value is a sub-slice of
+					// whole http body; copy it
+					// in case ingest is delayed
+					value := make([]byte, len(tag.Value))
+					copy(value, tag.Value)
```
Comment (pkalliok-aiven) [inline src/query/api/v1/handler/influxdb/write.go:172]: as discussed, this should be a good one for upstreaming

File: src/x/pool/object.go [hunk 2]
```diff
@@ -51,10 +51,12 @@ type objectPool struct {
 	refillHighWatermark int
 	filling             int32
 	initialized         int32
+	inUse               int32
```
Comment (pkalliok-aiven) [inline src/x/pool/object.go:54]: and this is a good candidate for upstreaming, I guess

File: src/query/api/v1/handler/prometheus/native/common.go [hunk 3]
```diff
@@ -55,6 +56,17 @@ const (
...
+	s := d.String()
+	if strings.HasSuffix(s, "m0s") {
+		s = s[:len(s)-2]
+	}
+	if strings.HasSuffix(s, "h0m") {
+		s = s[:len(s)-2]
+	}
+	return s
+}
+
```
Comment (pkalliok-aiven) [inline src/query/api/v1/handler/prometheus/native/common.go:69]: This should be checked whether it's still needed at all and maybe dropped (but in later PR)

File: src/query/api/v1/handler/prometheus/native/common.go [hunk 4]
```diff
@@ -98,6 +110,19 @@ func parseParams(
...
+	if v := fetchOpts.LookbackDuration; v != nil {
+		params.LookbackDuration = *v
+	}
+
+	// Grafana lack of alert rule template variable expansion handling here (sigh)
+	// https://github.com/grafana/grafana/issues/6557
+	query = strings.ReplaceAll(query, "$__interval", shortDur(params.Step))
+	query = strings.ReplaceAll(query, "$__range", shortDur(params.LookbackDuration))
+	// Grafana 7.2 adds $__rate_interval which is still not expanded in alerting. According to documentation it is <= 4x step
+	query = strings.ReplaceAll(query, "$__rate_interval", shortDur(time.Duration(4)*params.Step))
```
Comment (pkalliok-aiven) [inline src/query/api/v1/handler/prometheus/native/common.go:124]: our alertmanager alerts should be fixed to drop these (I guess).  Or, maybe upstream could be convinced that having the macros is actually a good idea.

**Review by pkalliok-aiven — APPROVED** [review]
looks good to me (and they've been reviewed earlier anyway)

**Review by safa-topal — APPROVED** [review]
:shipit:

**Pull request merged**